src/daemon/gui/gui.rs

```diff
@@ -0,0 +1,219 @@
+use std::sync::MutexGuard;
+
+use anyhow::Context;
+use gtk4::{Align, EventSequenceState, Fixed, FlowBox, Frame, gdk_pixbuf, GestureClick, gio::File, glib::clone, IconLookupFlags, IconPaintable, IconTheme, Label, Overflow, Overlay, pango, Picture, prelude::*, TextDirection};
+use hyprland::data::{Client, WorkspaceBasic};
+use log::{info, warn};
+
+use crate::{Share, SharedData};
+use crate::daemon::funcs::{close, switch_gui, switch_gui_workspace};
+use crate::daemon::gui::{ICON_SCALE, ICON_SIZE, icons, SIZE_FACTOR};
+
+fn client_ui(client: &Client, client_active: bool, show_title: bool, index: i32, enabled: bool, max_switch_offset: u8) -> Frame {
+    let theme = IconTheme::new();
+    let icon = if theme.has_icon(&client.class) {
+        // debug!("Icon found for {}", client.class);
+        theme.lookup_icon(
+            &client.class,
+            &[],
+            *ICON_SIZE,
+            *ICON_SCALE,
+            TextDirection::None,
+            IconLookupFlags::PRELOAD,
+        )
+    } else {
+        // debug!("Icon not found for {}", client.class);
```

Comment:
I notice you're commenting debug a lot here, I assume because these are temporary?
I'm pretty new to Rust but can't you use the `trace` macro here? They seem pretty useful logs to have either way.

Love the project btw, very useful application :)

Reply:
Im planing to rework where debug!, info!, warn!, etc is used as using -vv is pretty much useless as you get way to much info but -v is not enough info.

It might be a good idea to use trace! for logs like `debug!("Clients data: {:?}", clients_data)`